lib/timers.js

```diff
@@ -696,6 +705,7 @@ function createImmediate(args, callback) {
 
 
 exports.clearImmediate = function(immediate) {
+  // TODO: also verify `immediate instanceof Immediate`?
```

Comment:
It only checks property names, which I think could be problematic if someone passes in something that isn't a value returned by `setImmediate()`? I've removed the comment for now anyway...